### DISCUSSION TITLE
Why form-data cost much more time than json base64 data when transfer image

### DISCUSSION THREAD

**Discussion by se7enXF** [fastapi/fastapi#8533] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# -*- coding: utf-8 *-*

import time
import cv2
import base64
import numpy as np
from pydantic import BaseModel, Field
from fastapi import FastAPI, Request, File


def base64_to_image(img_bs64: str) -> np.ndarray:
    byte_image = base64.b64decode(img_bs64)
    np_image = np.frombuffer(byte_image, np.uint8)
    bgr_image = cv2.imdecode(np_image, cv2.IMREAD_COLOR)
    return bgr_image


class RequestBody(BaseModel):
    image: str = Field(...)


app = FastAPI()


@app.post('/form-data')
async def form_data(request: Request, image: bytes = File(...)):
    t1 = time.time()
    img_buffer = np.frombuffer(image, dtype=np.uint8)
    bgr_image = cv2.imdecode(img_buffer, cv2.IMREAD_COLOR)
    t2 = time.time()
    string = f'API: {request.url.path}, Shape: {bgr_image.shape}, process time: {t2 - t1}'
    print(string, flush=True)
    return string


@app.post('/base64-data')
async def base64_data(request: Request, body: RequestBody):
    t1 = time.time()
    bgr_image = base64_to_image(body.image)
    t2 = time.time()
    string = f'API: {request.url.path}, Shape: {bgr_image.shape}, process time: {t2 - t1}'
    print(string, flush=True)
    return string
```

### Description

* Example image shape: (1080, 1920, 3)
1.  Using postman to post image to `http://[IP]`
    - `/base64-data` cost 50 ms
    - `/form-data` cost 650 ms

2.  Using postman to post image to `http://127.0.0.1`
    - `/base64-data` cost 24 ms
    - `/form-data` cost 23 ms

2.  Using postman to post image to `http://localhost`
    - `/base64-data` cost 1054 ms
    - `/form-data` cost 1656 ms

Why using IP as HOST cost too much time in form-data transfer?
I use python scripts to simulate requests, get the same time cost like postman.

### Operating System

Windows

### FastAPI Version

0.76.0

### Python Version

3.7.5

**Comment by iudeen** [comment]
I see a difference in logic where you convert to  string. Can you try to not do the extra processing and measure for a fairer result?
